### ISSUE TITLE
`ValueError` when accessing second level Numpy objects in Help pane

### ISSUE THREAD

**Issue by being-legen9** [spyder-ide/spyder#21148]
### What steps will reproduce the problem?

when you go to help in environment console and in object you type libraryname.sublibraryname (Example: numpy.where in my case) it prompts error

### Traceback
```python-traceback
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder\plugins\help\widgets.py", line 906, in force_refresh
    self.set_object_text(None, force_refresh=True)
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder\plugins\help\widgets.py", line 936, in set_object_text
    found = self.show_help(text, ignore_unknown=ignore_unknown)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder\plugins\help\widgets.py", line 1060, in show_help
    doc = shell.get_doc(obj_text)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\help.py", line 140, in get_doc
    return self.call_kernel(blocking=True).get_doc(objtxt)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 557, in __call__
    return self._comms_wrapper._get_call_return_value(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 224, in _get_call_return_value
    return super(KernelComm, self)._get_call_return_value(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 440, in _get_call_return_value
    return self._sync_error(reply['value'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 496, in _sync_error
    error_wrapper.raise_error()
  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 90, in raise_error
    raise self.etype(self)
Exception in comms call get_doc:

  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 343, in _handle_remote_call
    return_value = self._remote_callback(
                   ^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\frontendcomm.py", line 273, in _remote_callback
    return super(FrontendComm, self)._remote_callback(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\comms\commbase.py", line 356, in _remote_callback
    return self._remote_call_handlers[call_name](
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\console\kernel.py", line 370, in get_doc
    return getdoc(obj)
           ^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\site-packages\spyder_kernels\utils\dochelpers.py", line 123, in getdoc
    sig = inspect.signature(obj)
          ^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\inspect.py", line 3279, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\inspect.py", line 3027, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\inspect.py", line 2519, in _signature_from_callable
    return _signature_from_builtin(sigcls, obj,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\admin\anaconda3\Lib\inspect.py", line 2326, in _signature_from_builtin
    raise ValueError("no signature found for builtin {!r}".format(func))

ValueError: no signature found for builtin <built-in function where>

```

## Versions

* Spyder version: 5.4.3  (conda)
* Python version: 3.11.3 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                             :  1.4.0 (OK)
chardet >=2.0.0                                  :  4.0.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  1.7.3 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.12.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.1 (OK)
jellyfish >=0.7                                  :  0.9.0 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  6.5.4 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
paramiko >=2.4.0                                 :  2.8.1 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.0 (OK)
pygments >=2.0                                   :  2.15.1 (OK)
pylint >=2.5.0,<3.0                              :  2.16.2 (OK)
pylint_venv >=2.1.1                              :  2.3.0 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.0.2 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.2 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.2.0 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  67.8.0 (OK)
sphinx >=0.6.6                                   :  5.0.2 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.3 (OK)
textdistance >=4.2.0                             :  4.2.1 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  2.1.6 (OK)
zmq >=22.1.0                                     :  25.1.0 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  3.7.1 (OK)
numpy >=1.7                                      :  1.24.3 (OK)
pandas >=1.1.1                                   :  1.5.3 (OK)
scipy >=0.17.0                                   :  1.10.1 (OK)
sympy >=0.7.3                                    :  1.11.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @being-legen9, thanks for reporting. I could easily reproduce this problem by following the instructions you posted above.

We'll fix it in our next version (5.5.0), to be released in a couple of months.

**Comment by being-legen9** [comment]
Thanks @ccordoba12 . I just wanted to say that this was the first time I have ever found out a bug. So can I please get some details whether it's severity was low or high ?

**Comment by being-legen9** [comment]
Hello @ccordoba12 , I tried fixing it but I am sorry I can't fix it.

**Comment by ccordoba12** [comment]
No worries, that's fine. We'll fix it in our next version (5.5.0), to be released in a couple of months.
